### PR TITLE
Create application cluster module

### PR DESCRIPTION
**Pull request by mattagohni**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  application_cluster_max_size = 2 # the maximum size of the cluster
+  application_cluster_min_size = 1 # the minimum size of the cluster
+  application_cluster_ssl_cert_arn = "${data.aws_acm_certificate.certificate.arn}"
+  application_cluster_user_data = "your-userdata-script-as-string" #can be loaded from a file
+  application_cluster_elb_access_log_bucket = "bucket-name-for-logging"
+  application_cluster_instance_role_id = "RoleIDForInstanceProfile"
+}
+
+data "aws_acm_certificate" "certificate" {
+  domain   = "${var.oms_domain}"
```
Comment (sd-lueckel) [inline README.md]: suggest not to use 'oms' in variable name

File: README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    + 1 Listener for Port 443 (An AWS-certificate is mandatory for this at the moment)
++ AutoScaling-Group
++ IAM-Instance-Profile
++ LaunchConfiguration
++ TargetGroup 
+
+## Usage
+### Parameters
+```hcl-terraform
+variable "application_cluster_elb_security_groups" {
```
Comment (sd-lueckel) [inline README.md]: suggest to use 'lb' instead of 'elb' because 'ApplicationLoadBalancer' != 'ElasticLoadBalancer'! Also used in other variables below

**Review by sd-lueckel — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -105,7 +105,7 @@ variable "application_cluster_user_data" {
 
 variable "application_cluster_ssl_policy" {
   description = "Policy for SSL"
-  default = "ELBSecurityPolicy-TLS-1-2-2017-01"
+  default = "lbSecurityPolicy-TLS-1-2-2017-01"
```
Comment (sd-lueckel) [inline README.md]: I'm not sure if this change was intended

File: variables.tf [hunk 4]
```diff
@@ -90,5 +90,5 @@ variable "application_cluster_user_data" {
 
 variable "application_cluster_ssl_policy" {
   description = "Policy for SSL"
-  default = "ELBSecurityPolicy-TLS-1-2-2017-01"
+  default = "lbSecurityPolicy-TLS-1-2-2017-01"
```
Comment (sd-lueckel) [inline variables.tf]: I'm not sure if this change was intended

**Review by sd-lueckel — CHANGES_REQUESTED** [review]

**Review by sd-lueckel — APPROVED** [review]

**Review by michz — APPROVED** [review]

**Pull request merged**